### ISSUE TITLE
[Assistive tech and device support] Keyboard navigation order isn't meaningful or sequential. (11.08.1)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#60733]
#### VFS team name
Digitize Forms

#### VFS product name
Form 21-10210 Lay/Witness Statement

#### Point of Contact/Reviewers
Brian DeConinck - @briandeconinck - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
Keyboard navigation order isn't meaningful or sequential.

### Issue Details
Keyboard focus management isn't behaving as expected for the first few screens of the workflow:
- When starting the form (going from /introduction to /claim-ownership), focus seems to land in different places depending on whether you're authenticated or unauthenticated.
- When pressing the Continue button to move to the next step, focus stays on the Continue button, meaning you have to [shift] + [tab] backwards to get back to the inputs.
- When pressing the Back button to move back through the workflow, focus stays on the Back button (same issue as Continue but in reverse).

Note that the odd focus management also impacts how screen readers announce the form. Without any visual cues, focus being placed somewhere unexpected means screen reader users may not be able to identify where they are or what has happened. For example, after clicking Continue, my screen reader doesn't announce any change in the content on the screen. I have no idea that I've moved to the next step and might think something is just broken.

### Link, screenshot or steps to recreate
Steps to reproduce:
1. As an unauthenticated user, navigate to /introduction.
2. Press [tab] to move focus to the "Start your application without signing in" and press [enter] to activate the link. Note that on the next page, the viewport is scrolled down about halfway.
3. Press [tab] to move to the next focusable element. Note that focus moves to the first radio button.
4. Go back to /introduction and log in with a staging test user.
5. Press [tab] to move focus to "Start your statement" and press [enter] to activate the link. Note that on the next page, the viewport is at the top of the page.
6. Press [tab] to move to the next focusable element. Note that focus moves to the first breadcrumb link.
7. Press [tab] to move to the first radio button and press [spacebar] to select the radio button.
8. Press [tab] until focus moves to the Continue button. Press [enter] to activate the button. Note that although you've moved to the next page, keyboard focus continues to be on the Continue button.
9. Press [shift] + [tab] to move focus to the Back button. Press [enter] to activate the button. Note that focus continues to be on the Back button.

### VA.gov Experience Standard
[Category Number 11, Issue Number 08](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 2.4.3_A

### Platform Recommendation
On most forms on VA.gov, focus moves to the H2 element in the step indicator. Since the step indicator is hidden for those first few questions, I'm guessing the scripts in the forms system that manage focus aren't working/are trying to target something that doesn't exist.

If possible, I'd like to see focus set to the H1 when you go to /claim-ownership, and then on the subsequent step focus set to the `legend`.

---

### VFS Guidance

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Platform help validating the issue, please comment on the ticket
- Some feedback provided may be out of scope for your iteration of the product, however, Platform's OCTO leadership has stated that all identified issues need to be documented and it is still your responsibility to resolve the issue.
- If you do not believe that this Staging Review issue ticket is the responsibility of your team, comment below providing an explanation and who you believe is responsible. Please tag the Point of Contact/Reviewers. Governance team will research and will follow up.

**Comment by tlei123** [comment]
Per Garrett, check customScrollAndFocus and focusElement [here](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/containers/FormPage.jsx#L26-L33).

**Comment by tlei123** [comment]
`[Bugfix PR](https://github.com/department-of-veterans-affairs/vets-website/pull/24666)` opened.

**Comment by tlei123** [comment]
Moving back to In Progress due to Firefox browser-specific issue.

**Comment by tlei123** [comment]
Issue fixed, fix tested against all Mac/Win browsers [incl. Edge], & [PR merged](https://github.com/department-of-veterans-affairs/vets-website/pull/24666).
